geoutils/georaster/raster.py

```diff
@@ -1663,17 +1669,18 @@ def get_bounds_projected(self, out_crs: CRS, densify_pts_max: int = 5000) -> rio
         # Calculate new bounds
         left, bottom, right, top = self.bounds
         new_bounds = rio.warp.transform_bounds(self.crs, out_crs, left, bottom, right, top, densify_pts)
+        new_bounds = rio.coords.BoundingBox(*new_bounds)
 
         return new_bounds
 
-    def intersection(self, rst: str | Raster) -> tuple[float, float, float, float]:
+    def intersection(self, rst: str | Raster, match_ref: bool = True) -> tuple[float, float, float, float]:
         """
         Returns the bounding box of intersection between this image and another.
 
         If the rasters have different projections, the intersection extent is given in self's projection system.
 
         :param rst : path to the second image (or another Raster instance)
-
+        :param match_ref: if set to True, returns the smallest intersection that is a multiple of self's georeferences
         :returns: extent of the intersection between the 2 images \
```

Comment:
Cannot understand "that is a multiple of self's georeferences" by itself. Is the "multiple" linked to the Raster's resolution?

Reply:
Done